base/applications/network/telnet/src/tnerror.cpp

```diff
@@ -92,19 +124,19 @@ int printm(LPTSTR szModule, BOOL fSystem, DWORD dwMessageId, ...)
 	va_list Ellipsis;
 	va_start(Ellipsis, dwMessageId);
 
-	LPTSTR pszMessage = 0;
+	LPWSTR wszMessage = 0;
 	DWORD dwMessage = 0;
 	if(fSystem) {
-		dwMessage = FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+		dwMessage = FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER |
 			FORMAT_MESSAGE_FROM_SYSTEM, hModule, dwMessageId,
-			LANG_USER_DEFAULT, (LPTSTR)&pszMessage, 128, &Ellipsis);
+			LANG_USER_DEFAULT, (LPWSTR)&wszMessage, 128, &Ellipsis);
 	} else {
 		// we will use a string table.
-		char szString[256];
-		if(LoadString(0, dwMessageId, szString, sizeof(szString)))
-			dwMessage = FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER |
-				FORMAT_MESSAGE_FROM_STRING, szString, dwMessageId,
-				LANG_USER_DEFAULT, (LPTSTR)&pszMessage, 256, &Ellipsis);
+		WCHAR wszString[256];
+		if(LoadStringW(0, dwMessageId, wszString, sizeof(wszString) / sizeof(*wszString)))
+			dwMessage = FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+				FORMAT_MESSAGE_FROM_STRING, wszString, dwMessageId,
+				LANG_USER_DEFAULT, (LPWSTR)&wszMessage, 256, &Ellipsis);
```

Comment:
(Note that this 256 is also `sizeof(wszString) / sizeof(*wszString)` .)

Reply:
Changed. Does the size matter here though, since `FORMAT_MESSAGE_ALLOCATE_BUFFER` is used? I didn't change it at first, because that's what was in the code originally - I don't really understand what the original author's intent was here, why it can't just be 0.

Reply:
You may have noticed that the telnet code is not of the highest standard, this may explain some things :joy:

Reply:
Yes, I have. IIRC I saw it assume `sizeof(TCHAR) == 1` somewhere. Yikes...